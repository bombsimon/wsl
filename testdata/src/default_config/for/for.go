package testpkg

func keepCuddledVar() {
	a := 1
	b := 2 // want `missing whitespace above this line \(too many statements above for\)`
	for i := 0; i < b; i++ {
		panic(1)
	}

	_, _ = a, b
}

func keepNone() {
	b := 2
	a := 1
	for i := 0; i < b; i++ { // want `missing whitespace above this line \(no shared variables above for\)`
		panic(1)
	}

	_, _ = a, b
}

func separateBlocks() {
	for i := 0; i < 1; i++ {
		panic("")
	}
	for i := 0; i < 1; i++ { // want `missing whitespace above this line \(invalid statement above for\)`
		panic("")
	}
}
