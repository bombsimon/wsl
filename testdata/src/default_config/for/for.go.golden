package testpkg

func fn1() {
	a := 1

	b := 2 // want `missing whitespace above this line \(too many statements above for\)`
	for i := 0; i < b; i++ {
		panic(1)
	}

	_ = a
	_ = b
}

func fn2() {
	b := 2
	a := 1

	for i := 0; i < b; i++ { // want `missing whitespace above this line \(no shared variables above for\)`
		panic(1)
	}

	_ = a
	_ = b
}

func fn3() {
	for i := 0; i < 1; i++ {
		panic("")
	}

	for i := 0; i < 1; i++ { // want `missing whitespace above this line \(invalid statement above for\)`
		panic("")
	}
}
